core/src/main/scala/kafka/log/Log.scala

```diff
@@ -543,6 +548,16 @@ class Log(@volatile var dir: File,
     }
   }
 
+  def renameDir(name: String): Boolean = {
+    val renamedDir = new File(dir.getParent, name)
+    val renameSuccessful = renamedDir == dir || dir.renameTo(renamedDir)
```

Comment:
`renamedDir` was instantiated in the line above, it can't possibly `== dir`. I think you mean `.equals()`

Reply:
`==` in Scala is _not_ reference equality.

Reply:
Doh! Thank you.